node/src/blockchain/blockchain_bridge.rs

```diff
@@ -1730,14 +1628,12 @@ mod tests {
             received_payments,
             &ReceivedPayments {
                 timestamp: received_payments.timestamp,
+                new_start_block: 8675309u64 + 1,
```

Comment:
I believe this is wrong. 
It should correspond **exactly** to the `new_start_block` in:
```
RetrievedBlockchainTransactions {
       new_start_block: 8675309u64,
       transactions: vec![BlockchainTransaction {
            block_number: 8675308u64,
            from: some_wallet.clone(),
            wei_amount: amount,
       }],
}
```
But you had to add one to make the assertion work and it only shows that the internal code is wrong. (We discussed the cause already. You will likely know by now how you should proceed)